DarkFlippers/unleashed-firmware
# fix[infrared]: fixed crash if button out of bounds and sending wrong signals

# What's fixed

There were two bugs:

1) If you select the last button of the TV remote (index 5), then go back to AC (has only 4 buttons, so most possible index 3) and press OK, the Flipper crashes.

2) The assignment of the buttons was not reset when changing the remote control. Therefore it could happen that if you first go to AC remote, then to Audio and press **Volume down** (`VOL-`), the Flipper tries to send **Temperature down** (`TEMP-`) (which obviously does not exist in Audio).

Thanks Amec0e and Nano for the help :)

# Verification 

- Select the last button in the TV remote, go back to AC, the first button shoud be selected
- Go to the AC remote, then to the Audio remote and confirm that sending works

# Checklist (For Reviewer)
